Series of small WCOSS2 updates - round 11

KateFriedman-NOAA opened the pull request:
**Description**

This PR folds in final updates from @WeiWei-NCO's ops install for gfs.v16.2:

1. update hand-off tag in release notes to `EMC-v16.2.0.4` (will cut tag after this PR goes in)
2. a number of resource adjustments based on NCO/EMC testing on Dogwood, particularly for the wave jobs
3. move the `excl` tag from it's own line in the ecf script PBS directives into the same line as the other resources settings (right after `vscatter`)
4. add `export OMP_PLACES=cores` to fcst block in `WCOSS2.env`
5. a few small script updates from Wei based on testing and errors
6. move transfer list parm files into new `transfer/` folder and update from NCO copies (bulk of the files changed in this PR)
6. update NETCDF flags in `gaussian_sfcanl.fd/makefile.sh` to match what is needed on R&D machines for correct compilation
7. updates to `run.ver`, including moving some version variables into the new `wcoss2.ver`
8. create `wcoss2.ver` to support developers who don't set a few version variables via ecflow because they will be running with rocoto (matches similar updates happening on dev side of this port)
9. update load_fv3gfs_modules.sh to source `wcoss2.ver`
10. update WAFS tag to `gfs_wafs.v6.2.8` (after final WAFS script edit was folded in)

Refs: #399

KateFriedman-NOAA committed:
Update compilation flags for gaussian_sfcanl build

- update the NETCDF_LDFLAGS_F flags in gaussian_sfcanl makefile.sh
- corrects the flags based on testing on R&D platforms

Refs: #399

KateFriedman-NOAA committed:
Add OMP_PLACES=cores for fcst block in WCOSS2.env

- use OMP_PLACES=cores setting based on optimization work for forecast
job on Dogwood

Refs: #399

KateFriedman-NOAA committed:
Update several versions in run.ver

- correct wrong w3emc_ver in run.ver to match version in build.ver
- per feedback from NCO update the craype_ver to 2.7.10 and
cray_pals_ver to 1.0.17

Refs: #399

KateFriedman-NOAA committed:
Update gfs_forecast job resources

- update the resource settings for the gfs_forecast job based on
optimization work on Dogwood

Refs: #399

KateFriedman-NOAA committed:
Move excl setting into resource line in ecf scripts

- per request from NCO have moved the excl setting from it's separate
line up into the main resource PBS directive line for jobs that run
exclusively

Refs: #399

KateFriedman-NOAA committed:
Script updates from NCO

- a few small adjustments to scripts for correctness on WCOSS2

Refs: #399

KateFriedman-NOAA committed:
Updates to run.ver and create wcoss2.ver

- ingest run.ver updates from NCO
- create wcoss2.ver to set versions removed from run.ver but needed to
support developers (e.g. prod_util)

Refs: #399

KateFriedman-NOAA committed:
Update EMC tag name in v16.2.0 release notes

- hand-off tag is now EMC-v16.2.0.4

Refs: #399

KateFriedman-NOAA committed:
Update wave job resources with NCO feedback

Refs: #399

KateFriedman-NOAA committed:
Move transfer lists into new transfer folder

- per NCO request created new transfer folder under /parm and moved all
transfer list parm files into new folder

Refs: #399

KateFriedman-NOAA committed:
Fold in transfer parm list updates from NCO

Refs: #399

KateFriedman-NOAA committed:
Updates to support wcoss2.ver

- add sourcing of wcoss2.ver to load_fv3gfs_modules.sh
- update wcoss2.ver to allow overrides from ecflow for prod_util_ver and
prod_envir_ver when run in ops

Refs: #399

KateFriedman-NOAA committed:
Update npe_node_fcst_gfs in config.resources.emc.dyn

- change matches update also done in config.resources.nco.static

Refs: #399

aerorahul reviewed (changes requested):
I don't agree with combining excl on the same line.

KateFriedman-NOAA commented:
> I don't agree with combining excl on the same line.

@WeiWei-NCO @aerorahul Please hash this out and tell me the final decision on where `excl` will be in the ecf. I've gotten two different instructions. Thanks!

WalterKolczynski-NOAA commented on parm/transfer/transfer_gdas_enkf_enkf_05.list:
```diff
@@ -24,8 +24,8 @@
 
-#com/gfs/_ENVIR_/enkfgdas._PDY_/
-/gpfs/dell1/nco/ops/com/gfs/_ENVIR_/enkfgdas._PDY_/
+#_COMROOT_/gfs/_SHORTVER_/enkfgdas._PDY_/
```
I know these are ops-only, but can we delete all these commented-out lines from these .list files? This one-liner should do it (but do it in a clean repo so you can reset if not):
`perl -i -p -e "s/^#_.*$//g *.list`

KateFriedman-NOAA replied:
Double checked with @WeiWei-NCO that those commented out lines weren't necessary and then removed them when he confirmed they weren't. Committed at f3ce093f.

WalterKolczynski-NOAA commented on scripts/exgfs_atmos_gempak_meta.sh:
```diff
@@ -62,7 +71,22 @@ do
 
-   awk '{print $1}' $FIXgempak/gfs_meta > $DATA/tmpscript
+   if [ $fhr -lt 100 ] ; then
+      typeset -Z2 fhr
```
1) This does not need to be in an if-block. If fhr > 100, it will simply do nothing.
2) This is a functional change from before, which zero-padded to three places. This is only zero-padding to two
3) If the if-block mattered and we kept the padding as-is, zero-padding would only be needed < 10.

KateFriedman-NOAA replied:
Will update in `develop` branch.

WalterKolczynski-NOAA commented on scripts/exgfs_atmos_gempak_meta.sh:
```diff
@@ -97,11 +126,12 @@ do
-  else
-    fhr=`expr $fhr + $fhinc`
-  fi
+      typeset -Z3 fhr
```
And now we zero-pad again, but to three places? There should probably be one of these at the very top of the loop, and that's it.

KateFriedman-NOAA replied:
Will update in `develop` branch.

WalterKolczynski-NOAA commented on scripts/exgfs_atmos_gempak_meta.sh:
```diff
@@ -97,11 +126,12 @@ do
-  fi
+      typeset -Z3 fhr
+      if [ $fhr -eq 126 ] ; then
+        let fhr=fhr+6
```
Use arithmetic expansion instead: `(( fhr = 10#fhr + 6 ))`
(The 10# is necessary because otherwise zero-padded numbers are treated as octals)

KateFriedman-NOAA replied:
Will update in `develop` branch.

WalterKolczynski-NOAA commented on scripts/exgfs_atmos_gempak_meta.sh:
```diff
@@ -97,11 +126,12 @@ do
+      if [ $fhr -eq 126 ] ; then
+        let fhr=fhr+6
+      else
+	let fhr=fhr+fhinc
```
Use arithmetic expansion instead

KateFriedman-NOAA replied:
Will update in `develop` branch.

WalterKolczynski-NOAA reviewed (dismissed):
A few things, plus the excl changes previously noted.

KateFriedman-NOAA committed:
Update WAFS tag to gfs_wafs.v6.2.8

Refs: #399

KateFriedman-NOAA committed:
Remove commented out lines from transfer lists

- remove the commented out lines from the enkf transfer lists which are
not needed
- did not touch comment sections at top

Refs: #399

KateFriedman-NOAA committed:
Move all PBS place settings to separate line

- move all place=vscatter and place=excl settings in ecf scripts to
their own line
- now have one line for PBS select information and one line for PBS
place information
- changes decided on in collaboration between NCO and EMC

Refs: #399

KateFriedman-NOAA commented:
Updated ecf script PBS directives with regard to `select` and `place` settings. After discussion between NCO and EMC the ecf scripts will now have one `select` line with those resource values and a separate `place` line for setting `vscatter` and `excl`. Example:

Prior (all on single line):
```
#PBS -l place=vscatter:excl,select=...
```
Now (two separate lines):
```
#PBS -l select=...
#PBS -l place=vscatter:excl
```

Committed @ 9a74a804.

KateFriedman-NOAA commented:
@BoiVuong-NOAA @WeiWei-NCO Please see Walter's comments above about the `scripts/exgfs_atmos_gempak_meta.sh` script. Thanks!

BoiVuong-NOAA commented:
Kate, 
Do I need to make changes as Walter's comments in the script scripts/exgfs_atmos_gempak_meta.sh  ? or leave it as is.

WeiWei-NCO commented:
> @BoiVuong-NOAA @WeiWei-NCO Please see Walter's comments above about the `scripts/exgfs_atmos_gempak_meta.sh` script. Thanks!

In #599 , Boi provided a fixed script. However, the new script does a lot more dbnet alerts comparing to WCOSS1. After discussed with Boi, I reverted the script back to the WCOSS1 version.

KateFriedman-NOAA commented:
> Kate, Do I need to make changes as Walter's comments in the script scripts/exgfs_atmos_gempak_meta.sh ? or leave it as is.

@BoiVuong-NOAA At this point I think it depends on whether @WeiWei-NCO is open to testing another updated copy of this script before we make our final tag for the GFS on WCOSS2. We definitely want to get this updated for the `develop` branch side of things (for future GFS implementations) but not sure if we have time to do this now. @WeiWei-NCO thoughts?

BoiVuong-NOAA commented:
NCO commended to fix this issue in GFS v17

WeiWei-NCO commented:
> > Kate, Do I need to make changes as Walter's comments in the script scripts/exgfs_atmos_gempak_meta.sh ? or leave it as is.
> 
> @BoiVuong-NOAA At this point I think it depends on whether @WeiWei-NCO is open to testing another updated copy of this script before we make our final tag for the GFS on WCOSS2. We definitely want to get this updated for the `develop` branch side of things (for future GFS implementations) but not sure if we have time to do this now. @WeiWei-NCO thoughts?

It depends on how soon we can get this fixed and tested...

Since it's working fine in ops for months, I am okay to leave it as is at this point. Please fix it in the future upgrade.

KateFriedman-NOAA commented:
> NCO commended to fix this issue in GFS v17

@BoiVuong-NOAA @WeiWei-NCO Thanks, we are agreed to make this change in the `develop` branch for GFSv17 but will leave the `exgfs_atmos_gempak_meta.sh` script as is for GFSv16.

KateFriedman-NOAA commented:
@WalterKolczynski-NOAA @aerorahul Any further comments on this PR? @WeiWei-NCO just let me know there may be another update coming in. I can merge this and submit an additional small PR for that change should it come in. Am waiting to tag until that additional change is sorted.

aerorahul reviewed (approved):
looks good.
